### PR TITLE
Fix cross-compile for Windows with CMake 3.12

### PR DESCRIPTION
**Pull request by redmanmale**
This is a copy of https://github.com/irungentoo/openal-soft-tox/pull/2 to the original repo (which probably wouldn't be merged anytime soon).

This change is needed to be able to build `openal` with modern CMake versions on modern Linux distros (Ubuntu Bionic) and also Cygwin on Windows.

Right now it's failing like this:

```
CMake Error at /usr/local/cmake-3.12.4/share/cmake-3.12/Modules/CheckIncludeFiles.cmake:63 (message):

  Unknown arguments:

    -D_WIN32_WINNT=0x0502

Call Stack (most recent call first):

  CMakeLists.txt:828 (CHECK_INCLUDE_FILES)

-- Configuring incomplete, errors occurred!
```

### CODE REVIEW

**Comment by redmanmale** [comment]
@robinlinden @avoidr
Could you take a look please.

**Comment by ghost** [comment]
As far as I can understand, our CMake build already complies with the new policy, but cmake only gives warning because the policy is not set. Is that right?

https://cmake.org/cmake/help/latest/policy/CMP0075.html

**Comment by redmanmale** [comment]
Our build uses Cmake 3.9.2

I've tried update our build to use new Cmake:
Failed: https://travis-ci.org/github/redmanmale/uTox/jobs/683013362
Success: https://travis-ci.org/github/redmanmale/uTox/jobs/683042048

**Comment by ghost** [comment]
Looks fine to me. @robinlinden?

File: CMakeLists.txt [hunk 1]
```diff
@@ -825,7 +828,7 @@ OPTION(ALSOFT_REQUIRE_DSOUND "Require DirectSound backend" OFF)
 OPTION(ALSOFT_REQUIRE_MMDEVAPI "Require MMDevApi backend" OFF)
 IF(HAVE_WINDOWS_H)
     # Check MMSystem backend
-    CHECK_INCLUDE_FILES("windows.h;mmsystem.h" HAVE_MMSYSTEM_H -D_WIN32_WINNT=0x0502)
+    CHECK_INCLUDE_FILES("windows.h;mmsystem.h" HAVE_MMSYSTEM_H)
```
Comment (robinlinden) [inline CMakeLists.txt:831]: This should be
```
set(CMAKE_REQUIRED_DEFINITIONS -D_WIN32_WINNT=0x0502)
CHECK_INCLUDE_FILES("windows.h;mmsystem.h" HAVE_MMSYSTEM_H)
unset(CMAKE_REQUIRED_DEFINITIONS)
```
if you want the same behaviour as before.

Comment (redmanmale) [reply CMakeLists.txt:831]: I can't fix it in the codedokode's branch.
I'll close this PR and open a new one with requested change.

**Comment by redmanmale** [comment]
Closing in favor of #2.

**Pull request closed without merging**